#!/bin/bash

### Command line arg $1 should be name of directory containing raw fastq files named with format:
###  L2_AGTCAA_L002_R1_001.fastq.gz
###	 L2_AGTCAA_L002_R2_001.fastq.gz
###  where "AGTCAA" is index for a subject 
###
### Adapter trimmed fastq files and separate files containing trimmed reads shorter than 50bp
###  will be written to directory $1

cd $1
SUBJECTS=`ls L2* | awk '{split($1,a,"_");print a[2]}' | uniq`
#SUBJECTS="AGTCAA CCGTCC CTTGTA GCCAAT GTGAAA"
for s in $SUBJECTS
do
	R1=`ls . | grep $s".*R1"`
	R2=`ls . | grep $s".*R2"`
	echo "--------------------------------------------------------"
 	echo "--------------  Working on "$s" ----------------"
 	echo "--------------------------------------------------------"
 	
 	echo "----------------  "$R1
	cutadapt -O 3 -e0.1 --too-short-output=$R1"_shortReadsWithAdapter_1" \
	-a "AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC"$s"ATCTCGTATGCCGTCTTCTGCTTG" -m 50 --paired-output "tmp.2.fastq" \
	-o "tmp.1.fastq" $R1 $R2
	
	echo "----------------  "$R2
	cutadapt -O 3 -e0.1 --too-short-output=$R2"_shortReadsWithAdapter_2" \
	-a AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT -m 50 --paired-output $R1".cutAdapters.fastq" \
	-o $R2".cutAdapters.fastq" "tmp.2.fastq" "tmp.1.fastq"
	
	#echo "----------------  moving files to /Volumes/fishstudies-1/_Elim_RNAseq/fastq/cutadapt/Mar2015/"
	#mv $R1".cutAdapters.fastq" "/Volumes/fishstudies-1/_Elim_RNAseq/fastq/cutadapt/Mar2015/"$R1".cutAdapters.fastq"
	#mv $R2".cutAdapters.fastq" "/Volumes/fishstudies-1/_Elim_RNAseq/fastq/cutadapt/Mar2015/"$R2".cutAdapters.fastq"
	#mv $R1"_shortReadsWithAdapter_1" "/Volumes/fishstudies-1/_Elim_RNAseq/fastq/cutadapt/Mar2015/"$R1"_shortReadsWithAdapter_1"
	#mv $R2"_shortReadsWithAdapter_2" "/Volumes/fishstudies-1/_Elim_RNAseq/fastq/cutadapt/Mar2015/"$R2"_shortReadsWithAdapter_2"
done

#########
#####

# bio-rdf02:Mar2015 burtonigenomics$ /Volumes/fishstudies/_scripts/run_cutadapt_in_dir.sh ~/Documents/_Elim_data/Mar2015/
# --------------------------------------------------------
# --------------  Working on CAGATC ----------------
# --------------------------------------------------------
# ----------------  L1_CAGATC_L001_R1_001.fastq.gz
# cutadapt version 1.4.2
# Command line parameters: -O 3 -e0.1 --too-short-output=L1_CAGATC_L001_R1_001.fastq.gz_shortReadsWithAdapter_1 -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCACCAGATCATCTCGTATGCCGTCTTCTGCTTG -m 50 --paired-output tmp.2.fastq -o tmp.1.fastq L1_CAGATC_L001_R1_001.fastq.gz L1_CAGATC_L001_R2_001.fastq.gz
# Maximum error rate: 10.00%
#    No. of adapters: 1
#    Processed reads:     46605506
#    Processed bases:   4660550600 bp (4660.6 Mbp)
#      Trimmed reads:      2787986 (6.0%)
#      Trimmed bases:     24891093 bp (24.9 Mbp) (0.53% of total)
#    Too short reads:        14495 (0.0% of processed reads)
#     Too long reads:            0 (0.0% of processed reads)
#         Total time:   1398.27 s
#      Time per read:      0.030 ms
# 
# === Adapter 1 ===
# 
# Adapter 'AGATCGGAAGAGCACACGTCTGAACTCCAGTCACCAGATCATCTCGTATGCCGTCTTCTGCTTG', length 64, was trimmed 2787986 times.
# 
# No. of allowed errors:
# 0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-39 bp: 3; 40-49 bp: 4; 50-59 bp: 5; 60-64 bp: 6
# 
# Overview of removed sequences
# length	count	expect	max.err	error counts
# 3	1052628	728211.0	0	1052628
# 4	293396	182052.8	0	293396
# 5	160175	45513.2	0	160175
# 6	119785	11378.3	0	119785
# 7	104920	2844.6	0	104920
# 8	95027	711.1	0	95027
# 9	89315	177.8	0	88763 552
# 10	83744	44.4	1	81220 2524
# 11	76546	11.1	1	74622 1924
# 12	69841	2.8	1	68389 1452
# 13	65803	0.7	1	64448 1355
# 14	61047	0.2	1	59766 1281
# 15	56230	0.0	1	55023 1207
# 16	49375	0.0	1	48286 1089
# 17	44823	0.0	1	43733 1090
# 18	38556	0.0	1	37593 907 56
# 19	33910	0.0	1	32778 1065 67
# 20	30206	0.0	2	28962 1116 128
# 21	27282	0.0	2	26121 1057 104
# 22	24364	0.0	2	23343 918 103
# 23	22108	0.0	2	21174 846 88
# 24	20421	0.0	2	19487 841 93
# 25	18557	0.0	2	17679 790 88
# 26	16526	0.0	2	15720 730 76
# 27	14971	0.0	2	14225 667 75 4
# 28	12894	0.0	2	12217 597 62 18
# 29	11207	0.0	2	10608 505 62 32
# 30	10122	0.0	3	9558 478 49 37
# 31	8617	0.0	3	8075 468 60 14
# 32	7681	0.0	3	7259 361 44 17
# 33	6656	0.0	3	6263 351 22 20
# 34	5850	0.0	3	5502 304 33 11
# 35	5420	0.0	3	5092 286 30 12
# 36	4887	0.0	3	4618 233 27 6 3
# 37	4390	0.0	3	4143 214 20 12 1
# 38	4054	0.0	3	3792 222 27 8 5
# 39	3651	0.0	3	3407 210 18 12 4
# 40	3076	0.0	4	2873 166 19 11 7
# 41	2582	0.0	4	2420 127 20 10 5
# 42	2177	0.0	4	1993 148 21 8 7
# 43	1801	0.0	4	1610 149 27 10 5
# 44	1581	0.0	4	1414 120 17 24 6
# 45	1426	0.0	4	1290 102 20 8 6
# 46	1374	0.0	4	1230 103 23 9 6 3
# 47	1216	0.0	4	1100 81 17 12 5 1
# 48	1219	0.0	4	1096 99 12 5 5 2
# 49	1108	0.0	4	1001 77 16 10 4
# 50	946	0.0	5	829 88 16 4 5 4
# 51	807	0.0	5	717 73 9 3 1 4
# 52	604	0.0	5	534 52 12 2 3 1
# 53	534	0.0	5	484 35 9 3 1 2
# 54	398	0.0	5	344 35 11 3 1 2 2
# 55	382	0.0	5	337 35 5 3 1 1
# 56	330	0.0	5	291 27 8 2 2
# 57	327	0.0	5	282 28 9 3 2 2 1
# 58	330	0.0	5	287 31 6 4 1 1
# 59	286	0.0	5	249 27 5 2 3
# 60	240	0.0	6	199 29 4 2 4 1 1
# 61	211	0.0	6	181 23 3 3 0 0 1
# 62	185	0.0	6	165 13 6 0 1
# 63	156	0.0	6	138 8 7 1 0 2
# 64	136	0.0	6	122 10 3 0 0 0 1
# 65	90	0.0	6	71 14 2 1 0 1 1
# 66	98	0.0	6	82 5 6 4 0 0 1
# 67	65	0.0	6	57 6 2
# 68	62	0.0	6	54 6 0 1 0 0 1
# 69	47	0.0	6	35 10 1 1
# 70	35	0.0	6	28 4 2 0 0 1
# 71	32	0.0	6	28 4
# 72	12	0.0	6	12
# 73	20	0.0	6	18 0 1 1
# 74	11	0.0	6	10 1
# 75	8	0.0	6	5 2 1
# 76	5	0.0	6	2 2 0 0 0 0 1
# 77	7	0.0	6	5 2
# 78	8	0.0	6	6 1 1
# 79	4	0.0	6	4
# 80	8	0.0	6	7 0 0 0 0 0 1
# 81	6	0.0	6	6
# 82	6	0.0	6	5 1
# 83	12	0.0	6	12
# 84	3	0.0	6	2 1
# 85	7	0.0	6	5 0 1 0 0 0 1
# 86	9	0.0	6	6 1 1 1
# 87	8	0.0	6	7 1
# 88	7	0.0	6	5 2
# 89	4	0.0	6	3 0 0 0 1
# 90	5	0.0	6	3 1 1
# 91	12	0.0	6	9 2 1
# 92	9	0.0	6	9
# 93	13	0.0	6	11 1 1
# 94	9	0.0	6	8 1
# 95	19	0.0	6	15 3 0 0 0 0 1
# 96	3	0.0	6	2 1
# 97	20	0.0	6	15 4 0 0 0 0 1
# 98	1	0.0	6	1
# 99	4	0.0	6	2 1 1
# 100	8900	0.0	6	794 6965 872 161 63 27 18
# 
# ----------------  L1_CAGATC_L001_R2_001.fastq.gz
# cutadapt version 1.4.2
# Command line parameters: -O 3 -e0.1 --too-short-output=L1_CAGATC_L001_R2_001.fastq.gz_shortReadsWithAdapter_2 -a AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT -m 50 --paired-output L1_CAGATC_L001_R1_001.fastq.gz.cutAdapters.fastq -o L1_CAGATC_L001_R2_001.fastq.gz.cutAdapters.fastq tmp.2.fastq tmp.1.fastq
# Maximum error rate: 10.00%
#    No. of adapters: 1
#    Processed reads:     46591011
#    Processed bases:   4659101100 bp (4659.1 Mbp)
#      Trimmed reads:      2844808 (6.1%)
#      Trimmed bases:     23877365 bp (23.9 Mbp) (0.51% of total)
#    Too short reads:           29 (0.0% of processed reads)
#     Too long reads:            0 (0.0% of processed reads)
#         Total time:   1354.05 s
#      Time per read:      0.029 ms
# 
# === Adapter 1 ===
# 
# Adapter 'AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT', length 58, was trimmed 2844808 times.
# 
# No. of allowed errors:
# 0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-39 bp: 3; 40-49 bp: 4; 50-58 bp: 5
# 
# Overview of removed sequences
# length	count	expect	max.err	error counts
# 3	1089523	727984.5	0	1089523
# 4	319352	181996.1	0	319352
# 5	170598	45499.0	0	170598
# 6	121800	11374.8	0	121800
# 7	105050	2843.7	0	105050
# 8	94804	710.9	0	94804
# 9	89387	177.7	0	88551 836
# 10	83623	44.4	1	81245 2378
# 11	76596	11.1	1	74254 2342
# 12	69653	2.8	1	68299 1354
# 13	65435	0.7	1	63827 1608
# 14	60816	0.2	1	59650 1166
# 15	55851	0.0	1	53923 1928
# 16	49073	0.0	1	47834 1239
# 17	44577	0.0	1	43436 1141
# 18	38190	0.0	1	36291 1868 31
# 19	33733	0.0	1	32680 1019 34
# 20	29993	0.0	2	28509 934 550
# 21	27092	0.0	2	25950 964 178
# 22	24284	0.0	2	23311 818 155
# 23	21991	0.0	2	21023 841 127
# 24	20314	0.0	2	19394 784 136
# 25	18436	0.0	2	17558 767 111
# 26	16451	0.0	2	15654 692 105
# 27	14876	0.0	2	14166 586 124
# 28	12809	0.0	2	12154 528 118 9
# 29	11129	0.0	2	10430 605 86 8
# 30	10083	0.0	3	9528 437 76 42
# 31	8565	0.0	3	7049 1268 136 112
# 32	7649	0.0	3	7074 312 74 189
# 33	6599	0.0	3	6010 416 108 65
# 34	5806	0.0	3	5318 368 82 38
# 35	5369	0.0	3	4952 327 67 23
# 36	4848	0.0	3	4489 290 52 17
# 37	4375	0.0	3	4035 283 41 16
# 38	4032	0.0	3	3701 262 54 14 1
# 39	3627	0.0	3	3350 215 44 15 3
# 40	3067	0.0	4	2826 188 26 16 11
# 41	2574	0.0	4	2360 162 32 10 10
# 42	2160	0.0	4	1977 143 19 10 11
# 43	1790	0.0	4	1541 183 41 12 13
# 44	1569	0.0	4	1428 107 20 9 5
# 45	1416	0.0	4	1291 86 20 13 6
# 46	1360	0.0	4	1179 89 62 17 12 1
# 47	1208	0.0	4	1059 95 25 15 9 5
# 48	1207	0.0	4	1051 96 18 21 15 6
# 49	1097	0.0	4	1001 60 19 8 9
# 50	942	0.0	5	839 73 17 6 5 2
# 51	1	0.0	5	1
# 52	2	0.0	5	0 0 0 0 0 2
# 53	4	0.0	5	2 0 1 0 1
# 54	2	0.0	5	1 0 0 0 1
# 55	2	0.0	5	1 0 0 0 1
# 56	2	0.0	5	0 1 0 1
# 57	1	0.0	5	0 0 0 1
# 58	2	0.0	5	1 1
# 59	2	0.0	5	0 0 0 1 1
# 61	1	0.0	5	0 0 0 1
# 64	1	0.0	5	0 0 0 0 0 1
# 65	1	0.0	5	0 1
# 66	1	0.0	5	1
# 67	1	0.0	5	0 0 0 1
# 68	2	0.0	5	0 2
# 70	2	0.0	5	0 0 1 0 1
# 100	2	0.0	5	0 1 0 1
# 
# ----------------  moving files to /Volumes/fishstudies-1/_Elim_RNAseq/fastq/cutadapt/Mar2015/
# --------------------------------------------------------
# --------------  Working on CCGTCC ----------------
# --------------------------------------------------------
# ----------------  L1_CCGTCC_L001_R1_001.fastq.gz
# cutadapt version 1.4.2
# Command line parameters: -O 3 -e0.1 --too-short-output=L1_CCGTCC_L001_R1_001.fastq.gz_shortReadsWithAdapter_1 -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCACCCGTCCATCTCGTATGCCGTCTTCTGCTTG -m 50 --paired-output tmp.2.fastq -o tmp.1.fastq L1_CCGTCC_L001_R1_001.fastq.gz L1_CCGTCC_L001_R2_001.fastq.gz
# Maximum error rate: 10.00%
#    No. of adapters: 1
#    Processed reads:     28777474
#    Processed bases:   2877747400 bp (2877.7 Mbp)
#      Trimmed reads:      1756469 (6.1%)
#      Trimmed bases:     16052696 bp (16.1 Mbp) (0.56% of total)
#    Too short reads:         9247 (0.0% of processed reads)
#     Too long reads:            0 (0.0% of processed reads)
#         Total time:    866.61 s
#      Time per read:      0.030 ms
# 
# === Adapter 1 ===
# 
# Adapter 'AGATCGGAAGAGCACACGTCTGAACTCCAGTCACCCGTCCATCTCGTATGCCGTCTTCTGCTTG', length 64, was trimmed 1756469 times.
# 
# No. of allowed errors:
# 0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-39 bp: 3; 40-49 bp: 4; 50-59 bp: 5; 60-64 bp: 6
# 
# Overview of removed sequences
# length	count	expect	max.err	error counts
# 3	636580	449648.0	0	636580
# 4	188540	112412.0	0	188540
# 5	103315	28103.0	0	103315
# 6	75752	7025.8	0	75752
# 7	66813	1756.4	0	66813
# 8	60896	439.1	0	60896
# 9	56664	109.8	0	56144 520
# 10	53506	27.4	1	51612 1894
# 11	49137	6.9	1	47745 1392
# 12	44719	1.7	1	43193 1526
# 13	42185	0.4	1	40792 1393
# 14	39045	0.1	1	37488 1557
# 15	35887	0.0	1	34415 1472
# 16	31588	0.0	1	30204 1384
# 17	28290	0.0	1	26955 1335
# 18	25112	0.0	1	23781 1173 158
# 19	22324	0.0	1	20957 1269 98
# 20	19986	0.0	2	18717 1113 156
# 21	18116	0.0	2	16994 991 131
# 22	16040	0.0	2	15035 914 91
# 23	14512	0.0	2	13587 822 103
# 24	13382	0.0	2	12493 795 94
# 25	12272	0.0	2	11450 721 101
# 26	11006	0.0	2	10211 724 71
# 27	9954	0.0	2	9267 620 60 7
# 28	8350	0.0	2	7772 499 62 17
# 29	7471	0.0	2	6907 475 61 28
# 30	6688	0.0	3	6148 457 53 30
# 31	5805	0.0	3	5265 444 75 21
# 32	5260	0.0	3	4827 346 66 21
# 33	4633	0.0	3	4223 352 43 15
# 34	4038	0.0	3	3668 313 47 10
# 35	3653	0.0	3	3352 258 36 7
# 36	3417	0.0	3	3117 250 35 13 2
# 37	3134	0.0	3	2880 203 37 10 4
# 38	2809	0.0	3	2507 264 19 14 5
# 39	2551	0.0	3	2309 200 19 6 17
# 40	2157	0.0	4	1949 123 11 61 13
# 41	1960	0.0	4	0 31 1794 65 70
# 42	1581	0.0	4	0 6 5 63 1507
# 43	1405	0.0	4	0 0 1257 105 43
# 44	1175	0.0	4	0 0 1015 118 42
# 45	1095	0.0	4	0 1 897 159 36 2
# 46	901	0.0	4	0 0 736 135 28 2
# 47	983	0.0	4	0 0 799 156 27 1
# 48	926	0.0	4	0 0 751 148 26 1
# 49	849	0.0	4	0 0 684 131 30 4
# 50	760	0.0	5	0 1 642 101 14 2
# 51	623	0.0	5	0 3 512 88 19 1
# 52	543	0.0	5	0 0 426 89 20 8
# 53	458	0.0	5	0 2 357 80 13 6
# 54	360	0.0	5	0 2 289 52 8 9
# 55	337	0.0	5	0 1 256 59 17 4
# 56	281	0.0	5	0 0 210 48 21 2
# 57	292	0.0	5	0 0 234 48 7 3
# 58	243	0.0	5	0 1 189 39 12 2
# 59	232	0.0	5	0 1 187 34 9 1
# 60	207	0.0	6	0 0 165 24 17 1
# 61	174	0.0	6	0 0 133 28 9 3 1
# 62	151	0.0	6	0 0 114 27 8 2
# 63	113	0.0	6	0 2 91 11 8 1
# 64	94	0.0	6	0 0 72 12 7 1 2
# 65	89	0.0	6	0 0 68 18 3
# 66	79	0.0	6	0 0 57 12 7 2 1
# 67	60	0.0	6	0 0 40 14 5 1
# 68	62	0.0	6	0 0 47 12 0 1 2
# 69	41	0.0	6	0 0 35 6
# 70	36	0.0	6	0 0 28 5 2 0 1
# 71	20	0.0	6	0 0 16 4
# 72	18	0.0	6	0 0 14 3 0 1
# 73	11	0.0	6	0 0 10 1
# 74	11	0.0	6	0 0 9 1 1
# 75	12	0.0	6	0 0 10 1 0 0 1
# 76	7	0.0	6	0 0 5 2
# 77	2	0.0	6	0 0 2
# 78	7	0.0	6	0 0 5 2
# 79	2	0.0	6	0 0 1 0 0 1
# 80	4	0.0	6	0 0 4
# 81	4	0.0	6	0 0 4
# 82	5	0.0	6	0 0 4 1
# 83	4	0.0	6	0 0 4
# 84	3	0.0	6	0 0 1 2
# 85	5	0.0	6	0 0 5
# 86	4	0.0	6	0 0 2 0 1 1
# 87	5	0.0	6	0 0 3 1 1
# 88	4	0.0	6	0 0 2 1 1
# 89	3	0.0	6	0 0 3
# 90	4	0.0	6	0 0 3 1
# 91	6	0.0	6	0 0 4 1 0 0 1
# 92	4	0.0	6	0 0 4
# 93	9	0.0	6	0 0 5 3 1
# 94	3	0.0	6	0 0 2 0 1
# 95	11	0.0	6	0 0 9 1 1
# 96	3	0.0	6	0 0 1 1 1
# 97	11	0.0	6	0 0 6 5
# 98	5	0.0	6	0 0 3 2
# 99	6	0.0	6	0 0 5 0 1
# 100	4579	0.0	6	0 1 403 3193 743 175 64
# 
# ----------------  L1_CCGTCC_L001_R2_001.fastq.gz
# cutadapt version 1.4.2
# Command line parameters: -O 3 -e0.1 --too-short-output=L1_CCGTCC_L001_R2_001.fastq.gz_shortReadsWithAdapter_2 -a AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT -m 50 --paired-output L1_CCGTCC_L001_R1_001.fastq.gz.cutAdapters.fastq -o L1_CCGTCC_L001_R2_001.fastq.gz.cutAdapters.fastq tmp.2.fastq tmp.1.fastq
# Maximum error rate: 10.00%
#    No. of adapters: 1
#    Processed reads:     28768227
#    Processed bases:   2876822700 bp (2876.8 Mbp)
#      Trimmed reads:      1840318 (6.4%)
#      Trimmed bases:     15594144 bp (15.6 Mbp) (0.54% of total)
#    Too short reads:           41 (0.0% of processed reads)
#     Too long reads:            0 (0.0% of processed reads)
#         Total time:    835.82 s
#      Time per read:      0.029 ms
# 
# === Adapter 1 ===
# 
# Adapter 'AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT', length 58, was trimmed 1840318 times.
# 
# No. of allowed errors:
# 0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-39 bp: 3; 40-49 bp: 4; 50-58 bp: 5
# 
# Overview of removed sequences
# length	count	expect	max.err	error counts
# 3	704284	449503.5	0	704284
# 4	207372	112375.9	0	207372
# 5	111261	28094.0	0	111261
# 6	77538	7023.5	0	77538
# 7	66910	1755.9	0	66910
# 8	60653	439.0	0	60653
# 9	56634	109.7	0	55977 657
# 10	53384	27.4	1	51769 1615
# 11	49243	6.9	1	47605 1638
# 12	44606	1.7	1	43700 906
# 13	41954	0.4	1	40919 1035
# 14	38866	0.1	1	38137 729
# 15	35594	0.0	1	34222 1372
# 16	31377	0.0	1	30503 874
# 17	28150	0.0	1	27442 708
# 18	24820	0.0	1	23469 1322 29
# 19	22098	0.0	1	21413 667 18
# 20	19782	0.0	2	18783 616 383
# 21	17956	0.0	2	17127 686 143
# 22	15946	0.0	2	15310 547 89
# 23	14441	0.0	2	13746 567 128
# 24	13302	0.0	2	12721 492 89
# 25	12186	0.0	2	11629 470 87
# 26	10930	0.0	2	10381 471 78
# 27	9890	0.0	2	9367 423 100
# 28	8277	0.0	2	7854 333 87 3
# 29	7412	0.0	2	6944 387 69 12
# 30	6656	0.0	3	6292 279 56 29
# 31	5754	0.0	3	4655 911 98 90
# 32	5226	0.0	3	4794 238 62 132
# 33	4593	0.0	3	4119 308 106 60
# 34	3991	0.0	3	3624 280 56 31
# 35	3627	0.0	3	3335 224 42 26
# 36	3384	0.0	3	3065 251 46 22
# 37	3101	0.0	3	2840 207 39 15
# 38	2780	0.0	3	2545 189 33 12 1
# 39	2528	0.0	3	2308 167 35 17 1
# 40	2143	0.0	4	1968 129 29 11 6
# 41	1958	0.0	4	1778 137 28 8 7
# 42	1587	0.0	4	1441 115 20 6 5
# 43	1400	0.0	4	1186 156 38 11 9
# 44	1172	0.0	4	1069 69 19 6 9
# 45	1097	0.0	4	993 73 15 12 4
# 46	910	0.0	4	795 61 37 9 8
# 47	981	0.0	4	874 78 18 8 3
# 48	922	0.0	4	828 61 11 15 6 1
# 49	844	0.0	4	773 53 9 3 5 1
# 50	757	0.0	5	692 49 7 5 1 3
# 51	9	0.0	5	1 2 3 1 1 1
# 53	6	0.0	5	2 3 0 0 1
# 54	3	0.0	5	0 0 1 0 2
# 55	3	0.0	5	0 0 2 0 0 1
# 56	3	0.0	5	0 0 1 0 2
# 57	3	0.0	5	0 0 1 0 0 2
# 58	1	0.0	5	0 1
# 59	1	0.0	5	0 1
# 61	2	0.0	5	2
# 63	1	0.0	5	0 1
# 64	1	0.0	5	0 0 0 0 1
# 71	1	0.0	5	0 0 0 0 1
# 82	1	0.0	5	0 0 0 1
# 100	6	0.0	5	2 2 0 0 0 2
# 
# ----------------  moving files to /Volumes/fishstudies-1/_Elim_RNAseq/fastq/cutadapt/Mar2015/
# --------------------------------------------------------
# --------------  Working on CGATGT ----------------
# --------------------------------------------------------
# ----------------  L1_CGATGT_L001_R1_001.fastq.gz
# cutadapt version 1.4.2
# Command line parameters: -O 3 -e0.1 --too-short-output=L1_CGATGT_L001_R1_001.fastq.gz_shortReadsWithAdapter_1 -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCACCGATGTATCTCGTATGCCGTCTTCTGCTTG -m 50 --paired-output tmp.2.fastq -o tmp.1.fastq L1_CGATGT_L001_R1_001.fastq.gz L1_CGATGT_L001_R2_001.fastq.gz
# Maximum error rate: 10.00%
#    No. of adapters: 1
#    Processed reads:     23803908
#    Processed bases:   2380390800 bp (2380.4 Mbp)
#      Trimmed reads:      1379331 (5.8%)
#      Trimmed bases:     12151645 bp (12.2 Mbp) (0.51% of total)
#    Too short reads:         7634 (0.0% of processed reads)
#     Too long reads:            0 (0.0% of processed reads)
#         Total time:    712.11 s
#      Time per read:      0.030 ms
# 
# === Adapter 1 ===
# 
# Adapter 'AGATCGGAAGAGCACACGTCTGAACTCCAGTCACCGATGTATCTCGTATGCCGTCTTCTGCTTG', length 64, was trimmed 1379331 times.
# 
# No. of allowed errors:
# 0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-39 bp: 3; 40-49 bp: 4; 50-59 bp: 5; 60-64 bp: 6
# 
# Overview of removed sequences
# length	count	expect	max.err	error counts
# 3	530321	371936.1	0	530321
# 4	151215	92984.0	0	151215
# 5	81581	23246.0	0	81581
# 6	59796	5811.5	0	59796
# 7	52112	1452.9	0	52112
# 8	46660	363.2	0	46660
# 9	43122	90.8	0	42837 285
# 10	39575	22.7	1	38384 1191
# 11	36482	5.7	1	35565 917
# 12	32802	1.4	1	32074 728
# 13	30388	0.4	1	29757 631
# 14	28729	0.1	1	28107 622
# 15	26072	0.0	1	25489 583
# 16	23260	0.0	1	22724 536
# 17	20722	0.0	1	20214 508
# 18	18060	0.0	1	17590 432 38
# 19	16181	0.0	1	15629 506 46
# 20	14298	0.0	2	13838 413 47
# 21	12640	0.0	2	12194 391 55
# 22	11277	0.0	2	10822 403 52
# 23	10417	0.0	2	10054 314 49
# 24	9487	0.0	2	9147 300 40
# 25	8768	0.0	2	8434 301 33
# 26	7976	0.0	2	7656 293 27
# 27	7121	0.0	2	6820 253 42 6
# 28	6298	0.0	2	6034 225 31 8
# 29	5311	0.0	2	5041 220 32 18
# 30	4870	0.0	3	4619 207 29 15
# 31	4167	0.0	3	3914 218 26 9
# 32	3667	0.0	3	3478 164 16 9
# 33	3233	0.0	3	3042 165 16 10
# 34	2895	0.0	3	2724 152 12 7
# 35	2756	0.0	3	2599 136 15 6
# 36	2441	0.0	3	2331 96 10 4
# 37	2372	0.0	3	2237 107 20 7 1
# 38	2291	0.0	3	2183 91 11 6
# 39	1922	0.0	3	1826 78 10 6 2
# 40	1661	0.0	4	1509 126 15 5 6
# 41	1482	0.0	4	1377 91 10 3 1
# 42	1152	0.0	4	1049 74 12 8 9
# 43	1072	0.0	4	991 58 14 9
# 44	884	0.0	4	799 60 6 13 6
# 45	833	0.0	4	761 59 11 1 1
# 46	698	0.0	4	652 42 2 1 1
# 47	717	0.0	4	653 43 8 5 8
# 48	735	0.0	4	665 60 8 1 1
# 49	649	0.0	4	590 52 4 3
# 50	529	0.0	5	489 34 3 2 1
# 51	502	0.0	5	453 44 2 2 0 1
# 52	418	0.0	5	377 28 5 3 1 4
# 53	330	0.0	5	299 25 3 1 2
# 54	263	0.0	5	239 18 4 0 0 2
# 55	211	0.0	5	194 9 2 2 2 0 2
# 56	196	0.0	5	170 19 2 0 2 3
# 57	186	0.0	5	171 10 4 0 0 0 1
# 58	157	0.0	5	139 12 4 1 1
# 59	188	0.0	5	164 18 5 0 0 1
# 60	138	0.0	6	118 16 2 0 0 1 1
# 61	111	0.0	6	101 6 2 2
# 62	107	0.0	6	99 6 1 1
# 63	97	0.0	6	84 8 1 3 0 1
# 64	53	0.0	6	48 4 0 1
# 65	59	0.0	6	54 3 1 0 1
# 66	54	0.0	6	49 3 1 1
# 67	37	0.0	6	30 3 1 1 0 2
# 68	21	0.0	6	17 3 0 0 1
# 69	20	0.0	6	17 1 0 0 1 0 1
# 70	22	0.0	6	19 1 1 1
# 71	13	0.0	6	11 1 1
# 72	16	0.0	6	13 2 1
# 73	9	0.0	6	7 1 1
# 74	4	0.0	6	3 0 1
# 75	6	0.0	6	4 2
# 76	6	0.0	6	5 1
# 77	7	0.0	6	4 1 1 1
# 78	8	0.0	6	8
# 79	1	0.0	6	0 0 0 0 0 1
# 80	2	0.0	6	2
# 81	2	0.0	6	2
# 82	2	0.0	6	2
# 83	3	0.0	6	2 0 0 0 0 1
# 84	3	0.0	6	3
# 85	3	0.0	6	3
# 86	2	0.0	6	1 1
# 87	4	0.0	6	4
# 88	4	0.0	6	4
# 89	4	0.0	6	4
# 90	2	0.0	6	1 0 0 1
# 91	2	0.0	6	2
# 92	4	0.0	6	3 1
# 93	4	0.0	6	4
# 94	5	0.0	6	4 0 0 1
# 95	3	0.0	6	3
# 96	1	0.0	6	1
# 97	13	0.0	6	13
# 99	1	0.0	6	1
# 100	4330	0.0	6	4 3787 400 102 27 6 4
# 
# ----------------  L1_CGATGT_L001_R2_001.fastq.gz
# cutadapt version 1.4.2
# Command line parameters: -O 3 -e0.1 --too-short-output=L1_CGATGT_L001_R2_001.fastq.gz_shortReadsWithAdapter_2 -a AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT -m 50 --paired-output L1_CGATGT_L001_R1_001.fastq.gz.cutAdapters.fastq -o L1_CGATGT_L001_R2_001.fastq.gz.cutAdapters.fastq tmp.2.fastq tmp.1.fastq
# Maximum error rate: 10.00%
#    No. of adapters: 1
#    Processed reads:     23796274
#    Processed bases:   2379627400 bp (2379.6 Mbp)
#      Trimmed reads:      1421737 (6.0%)
#      Trimmed bases:     11668415 bp (11.7 Mbp) (0.49% of total)
#    Too short reads:           12 (0.0% of processed reads)
#     Too long reads:            0 (0.0% of processed reads)
#         Total time:    688.42 s
#      Time per read:      0.029 ms
# 
# === Adapter 1 ===
# 
# Adapter 'AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT', length 58, was trimmed 1421737 times.
# 
# No. of allowed errors:
# 0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-39 bp: 3; 40-49 bp: 4; 50-58 bp: 5
# 
# Overview of removed sequences
# length	count	expect	max.err	error counts
# 3	562002	371816.8	0	562002
# 4	165703	92954.2	0	165703
# 5	86569	23238.5	0	86569
# 6	61006	5809.6	0	61006
# 7	52137	1452.4	0	52137
# 8	46509	363.1	0	46509
# 9	43346	90.8	0	42684 662
# 10	39499	22.7	1	38333 1166
# 11	36463	5.7	1	35326 1137
# 12	32705	1.4	1	32010 695
# 13	30203	0.4	1	29468 735
# 14	28565	0.1	1	28036 529
# 15	25833	0.0	1	24935 898
# 16	23065	0.0	1	22498 567
# 17	20549	0.0	1	20034 515
# 18	17848	0.0	1	16984 851 13
# 19	16026	0.0	1	15601 415 10
# 20	14180	0.0	2	13462 435 283
# 21	12573	0.0	2	12021 461 91
# 22	11205	0.0	2	10777 367 61
# 23	10345	0.0	2	9920 348 77
# 24	9421	0.0	2	9041 318 62
# 25	8714	0.0	2	8314 352 48
# 26	7922	0.0	2	7588 287 47
# 27	7050	0.0	2	6748 250 52
# 28	6244	0.0	2	5938 249 55 2
# 29	5279	0.0	2	4970 256 46 7
# 30	4847	0.0	3	4619 168 42 18
# 31	4139	0.0	3	3420 597 60 62
# 32	3633	0.0	3	3358 139 32 104
# 33	3204	0.0	3	2876 222 73 33
# 34	2873	0.0	3	2648 167 40 18
# 35	2745	0.0	3	2531 171 27 16
# 36	2428	0.0	3	2219 163 31 15
# 37	2353	0.0	3	2183 134 24 11 1
# 38	2277	0.0	3	2094 153 20 9 1
# 39	1904	0.0	3	1762 118 20 3 1
# 40	1648	0.0	4	1524 96 18 6 4
# 41	1484	0.0	4	1355 104 13 11 1
# 42	1151	0.0	4	1038 85 19 4 5
# 43	1066	0.0	4	926 115 15 4 6
# 44	878	0.0	4	794 58 19 5 2
# 45	828	0.0	4	756 47 12 5 8
# 46	695	0.0	4	608 38 35 10 4
# 47	714	0.0	4	643 46 18 3 4
# 48	727	0.0	4	665 37 10 7 4 4
# 49	643	0.0	4	572 49 12 5 2 3
# 50	527	0.0	5	484 26 9 3 4 1
# 51	3	0.0	5	1 0 0 1 0 1
# 58	1	0.0	5	0 0 0 0 0 1
# 59	1	0.0	5	0 0 0 1
# 60	1	0.0	5	0 0 1
# 64	1	0.0	5	0 1
# 67	1	0.0	5	0 0 0 0 1
# 70	3	0.0	5	0 0 1 2
# 73	1	0.0	5	0 0 0 0 0 1
# 
# ----------------  moving files to /Volumes/fishstudies-1/_Elim_RNAseq/fastq/cutadapt/Mar2015/
# --------------------------------------------------------
# --------------  Working on GCCAAT ----------------
# --------------------------------------------------------
# ----------------  L1_GCCAAT_L001_R1_001.fastq.gz
# cutadapt version 1.4.2
# Command line parameters: -O 3 -e0.1 --too-short-output=L1_GCCAAT_L001_R1_001.fastq.gz_shortReadsWithAdapter_1 -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCACGCCAATATCTCGTATGCCGTCTTCTGCTTG -m 50 --paired-output tmp.2.fastq -o tmp.1.fastq L1_GCCAAT_L001_R1_001.fastq.gz L1_GCCAAT_L001_R2_001.fastq.gz
# Maximum error rate: 10.00%
#    No. of adapters: 1
#    Processed reads:     34383535
#    Processed bases:   3438353500 bp (3438.4 Mbp)
#      Trimmed reads:      2330259 (6.8%)
#      Trimmed bases:     22341949 bp (22.3 Mbp) (0.65% of total)
#    Too short reads:         9918 (0.0% of processed reads)
#     Too long reads:            0 (0.0% of processed reads)
#         Total time:   1034.81 s
#      Time per read:      0.030 ms
# 
# === Adapter 1 ===
# 
# Adapter 'AGATCGGAAGAGCACACGTCTGAACTCCAGTCACGCCAATATCTCGTATGCCGTCTTCTGCTTG', length 64, was trimmed 2330259 times.
# 
# No. of allowed errors:
# 0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-39 bp: 3; 40-49 bp: 4; 50-59 bp: 5; 60-64 bp: 6
# 
# Overview of removed sequences
# length	count	expect	max.err	error counts
# 3	768920	537242.7	0	768920
# 4	235117	134310.7	0	235117
# 5	136381	33577.7	0	136381
# 6	104304	8394.4	0	104304
# 7	92490	2098.6	0	92490
# 8	85517	524.7	0	85517
# 9	79569	131.2	0	79137 432
# 10	75833	32.8	1	73696 2137
# 11	69534	8.2	1	67945 1589
# 12	64870	2.0	1	63495 1375
# 13	61498	0.5	1	60302 1196
# 14	56991	0.1	1	55781 1210
# 15	52960	0.0	1	51709 1251
# 16	47322	0.0	1	46185 1137
# 17	42319	0.0	1	41282 1037
# 18	37141	0.0	1	36188 896 57
# 19	33257	0.0	1	32090 1090 77
# 20	30053	0.0	2	29001 932 120
# 21	27099	0.0	2	26136 857 106
# 22	24147	0.0	2	23276 788 83
# 23	22235	0.0	2	21479 675 81
# 24	20469	0.0	2	19732 660 77
# 25	18704	0.0	2	17977 649 78
# 26	16539	0.0	2	15809 656 74
# 27	14945	0.0	2	14347 536 59 3
# 28	12558	0.0	2	12048 440 57 13
# 29	11277	0.0	2	10742 441 53 41
# 30	10026	0.0	3	9511 421 41 53
# 31	8554	0.0	3	8043 425 60 26
# 32	7633	0.0	3	7250 315 45 23
# 33	6684	0.0	3	6298 322 49 15
# 34	6005	0.0	3	5696 272 26 11
# 35	5279	0.0	3	4962 272 30 15
# 36	4819	0.0	3	4544 242 24 9
# 37	4334	0.0	3	4114 183 31 6
# 38	3865	0.0	3	3648 193 18 6
# 39	3483	0.0	3	3297 154 21 8 3
# 40	2921	0.0	4	2739 144 20 15 3
# 41	2545	0.0	4	2392 121 25 4 3
# 42	2136	0.0	4	1970 136 18 6 6
# 43	1774	0.0	4	1624 116 26 4 4
# 44	1510	0.0	4	1370 101 21 14 4
# 45	1362	0.0	4	1243 97 18 3 1
# 46	1187	0.0	4	1100 72 12 0 1 2
# 47	1208	0.0	4	1096 83 9 9 10 1
# 48	1110	0.0	4	1019 65 20 4 2
# 49	1025	0.0	4	946 61 11 3 4
# 50	832	0.0	5	756 58 8 6 2 2
# 51	737	0.0	5	665 50 12 2 6 2
# 52	573	0.0	5	515 49 6 1 1 1
# 53	455	0.0	5	411 26 12 4 2
# 54	408	0.0	5	362 27 12 1 1 1 4
# 55	322	0.0	5	278 27 4 6 3 2 2
# 56	278	0.0	5	251 19 3 1 1 1 2
# 57	322	0.0	5	282 32 5 0 2 1
# 58	301	0.0	5	262 32 6 1
# 59	275	0.0	5	243 25 2 1 2 2
# 60	256	0.0	6	224 17 7 3 3 2
# 61	177	0.0	6	163 11 3
# 62	170	0.0	6	147 18 3 2
# 63	128	0.0	6	112 10 1 4 0 0 1
# 64	85	0.0	6	74 8 1 2
# 65	83	0.0	6	72 10 1
# 66	78	0.0	6	69 7 1 1
# 67	55	0.0	6	48 4 2 0 0 0 1
# 68	55	0.0	6	51 2 2
# 69	41	0.0	6	39 2
# 70	29	0.0	6	26 2 0 1
# 71	21	0.0	6	18 0 1 1 1
# 72	24	0.0	6	22 1 0 1
# 73	8	0.0	6	8
# 74	6	0.0	6	5 0 0 0 1
# 75	8	0.0	6	5 1 2
# 76	5	0.0	6	4 1
# 77	8	0.0	6	7 1
# 78	4	0.0	6	3 1
# 79	4	0.0	6	4
# 80	6	0.0	6	4 1 0 0 1
# 81	5	0.0	6	5
# 82	5	0.0	6	5
# 83	6	0.0	6	5 1
# 84	3	0.0	6	3
# 85	2	0.0	6	1 1
# 86	5	0.0	6	3 2
# 87	3	0.0	6	3
# 88	3	0.0	6	3
# 89	5	0.0	6	5
# 90	7	0.0	6	4 3
# 91	5	0.0	6	5
# 92	4	0.0	6	4
# 93	8	0.0	6	6 1 0 1
# 94	6	0.0	6	6
# 95	16	0.0	6	15 1
# 96	11	0.0	6	10 1
# 97	13	0.0	6	11 1 0 1
# 98	7	0.0	6	6 1
# 99	4	0.0	6	3 1
# 100	4878	0.0	6	230 4059 450 96 24 14 5
# 
# ----------------  L1_GCCAAT_L001_R2_001.fastq.gz
# cutadapt version 1.4.2
# Command line parameters: -O 3 -e0.1 --too-short-output=L1_GCCAAT_L001_R2_001.fastq.gz_shortReadsWithAdapter_2 -a AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT -m 50 --paired-output L1_GCCAAT_L001_R1_001.fastq.gz.cutAdapters.fastq -o L1_GCCAAT_L001_R2_001.fastq.gz.cutAdapters.fastq tmp.2.fastq tmp.1.fastq
# Maximum error rate: 10.00%
#    No. of adapters: 1
#    Processed reads:     34373617
#    Processed bases:   3437361700 bp (3437.4 Mbp)
#      Trimmed reads:      2426706 (7.1%)
#      Trimmed bases:     21846428 bp (21.8 Mbp) (0.64% of total)
#    Too short reads:           14 (0.0% of processed reads)
#     Too long reads:            0 (0.0% of processed reads)
#         Total time:   1002.13 s
#      Time per read:      0.029 ms
# 
# === Adapter 1 ===
# 
# Adapter 'AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT', length 58, was trimmed 2426706 times.
# 
# No. of allowed errors:
# 0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-39 bp: 3; 40-49 bp: 4; 50-58 bp: 5
# 
# Overview of removed sequences
# length	count	expect	max.err	error counts
# 3	846106	537087.8	0	846106
# 4	258265	134271.9	0	258265
# 5	146092	33568.0	0	146092
# 6	106211	8392.0	0	106211
# 7	92419	2098.0	0	92419
# 8	85024	524.5	0	85024
# 9	79616	131.1	0	78673 943
# 10	75493	32.8	1	73437 2056
# 11	69461	8.2	1	67360 2101
# 12	64664	2.0	1	63359 1305
# 13	61126	0.5	1	59545 1581
# 14	56694	0.1	1	55551 1143
# 15	52477	0.0	1	50483 1994
# 16	46934	0.0	1	45692 1242
# 17	42033	0.0	1	40944 1089
# 18	36681	0.0	1	34660 1974 47
# 19	32982	0.0	1	31970 975 37
# 20	29814	0.0	2	28218 979 617
# 21	26906	0.0	2	25612 1069 225
# 22	23992	0.0	2	23007 816 169
# 23	22077	0.0	2	21112 820 145
# 24	20363	0.0	2	19422 826 115
# 25	18570	0.0	2	17631 809 130
# 26	16442	0.0	2	15555 750 137
# 27	14832	0.0	2	14070 629 132 1
# 28	12471	0.0	2	11800 555 107 9
# 29	11191	0.0	2	10436 630 105 20
# 30	9981	0.0	3	9439 418 82 42
# 31	8498	0.0	3	6798 1393 146 161
# 32	7583	0.0	3	6944 334 101 204
# 33	6625	0.0	3	5953 439 146 87
# 34	5959	0.0	3	5395 418 95 51
# 35	5242	0.0	3	4757 359 93 33
# 36	4780	0.0	3	4354 353 56 17
# 37	4304	0.0	3	3945 289 55 14 1
# 38	3834	0.0	3	3498 274 43 19
# 39	3456	0.0	3	3164 221 47 20 4
# 40	2896	0.0	4	2653 192 34 12 5
# 41	2530	0.0	4	2290 174 42 17 7
# 42	2121	0.0	4	1938 137 26 13 7
# 43	1765	0.0	4	1529 177 37 10 12
# 44	1498	0.0	4	1359 97 21 10 11
# 45	1350	0.0	4	1201 101 17 16 15
# 46	1183	0.0	4	995 91 64 18 13 2
# 47	1201	0.0	4	1062 81 29 19 10
# 48	1100	0.0	4	967 79 23 15 9 7
# 49	1020	0.0	4	906 76 21 9 7 1
# 50	830	0.0	5	728 52 26 9 10 5
# 53	2	0.0	5	1 0 0 1
# 54	1	0.0	5	1
# 55	3	0.0	5	2 0 1
# 56	1	0.0	5	1
# 57	1	0.0	5	1
# 59	2	0.0	5	0 2
# 60	1	0.0	5	0 1
# 64	1	0.0	5	0 0 0 0 0 1
# 68	1	0.0	5	0 0 0 0 1
# 79	1	0.0	5	0 1
# 
# ----------------  moving files to /Volumes/fishstudies-1/_Elim_RNAseq/fastq/cutadapt/Mar2015/
# --------------------------------------------------------
# --------------  Working on TGACCA ----------------
# --------------------------------------------------------
# ----------------  L1_TGACCA_L001_R1_001.fastq.gz
# cutadapt version 1.4.2
# Command line parameters: -O 3 -e0.1 --too-short-output=L1_TGACCA_L001_R1_001.fastq.gz_shortReadsWithAdapter_1 -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCACTGACCAATCTCGTATGCCGTCTTCTGCTTG -m 50 --paired-output tmp.2.fastq -o tmp.1.fastq L1_TGACCA_L001_R1_001.fastq.gz L1_TGACCA_L001_R2_001.fastq.gz
# Maximum error rate: 10.00%
#    No. of adapters: 1
#    Processed reads:     21260579
#    Processed bases:   2126057900 bp (2126.1 Mbp)
#      Trimmed reads:      1262951 (5.9%)
#      Trimmed bases:     11111371 bp (11.1 Mbp) (0.52% of total)
#    Too short reads:         5991 (0.0% of processed reads)
#     Too long reads:            0 (0.0% of processed reads)
#         Total time:    636.41 s
#      Time per read:      0.030 ms
# 
# === Adapter 1 ===
# 
# Adapter 'AGATCGGAAGAGCACACGTCTGAACTCCAGTCACTGACCAATCTCGTATGCCGTCTTCTGCTTG', length 64, was trimmed 1262951 times.
# 
# No. of allowed errors:
# 0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-39 bp: 3; 40-49 bp: 4; 50-59 bp: 5; 60-64 bp: 6
# 
# Overview of removed sequences
# length	count	expect	max.err	error counts
# 3	474110	332196.5	0	474110
# 4	137686	83049.1	0	137686
# 5	74918	20762.3	0	74918
# 6	55857	5190.6	0	55857
# 7	48623	1297.6	0	48623
# 8	44030	324.4	0	44030
# 9	40387	81.1	0	40099 288
# 10	37806	20.3	1	36662 1144
# 11	34091	5.1	1	33136 955
# 12	31337	1.3	1	30587 750
# 13	29142	0.3	1	28530 612
# 14	26932	0.1	1	26341 591
# 15	25077	0.0	1	24457 620
# 16	22310	0.0	1	21725 585
# 17	19920	0.0	1	19384 536
# 18	17222	0.0	1	16749 448 25
# 19	15142	0.0	1	14633 481 28
# 20	13456	0.0	2	12950 450 56
# 21	12215	0.0	2	11778 386 51
# 22	10678	0.0	2	10250 375 53
# 23	9538	0.0	2	9202 299 37
# 24	8809	0.0	2	8462 317 30
# 25	8017	0.0	2	7672 300 45
# 26	7506	0.0	2	7207 276 23
# 27	6461	0.0	2	6189 243 29
# 28	5592	0.0	2	5362 196 27 7
# 29	4884	0.0	2	4665 180 22 17
# 30	4302	0.0	3	4100 169 15 18
# 31	3933	0.0	3	3724 179 24 6
# 32	3344	0.0	3	3167 136 30 11
# 33	2918	0.0	3	2766 136 13 3
# 34	2582	0.0	3	2452 115 10 5
# 35	2300	0.0	3	2171 116 10 3
# 36	2033	0.0	3	1940 76 8 9
# 37	1931	0.0	3	1830 85 9 6 1
# 38	1836	0.0	3	1730 93 6 5 2
# 39	1550	0.0	3	1477 55 14 3 1
# 40	1436	0.0	4	1373 52 6 4 1
# 41	1225	0.0	4	1149 67 5 4
# 42	972	0.0	4	887 70 4 3 8
# 43	901	0.0	4	781 100 15 4 1
# 44	683	0.0	4	590 77 6 5 5
# 45	688	0.0	4	594 79 13 1 0 1
# 46	589	0.0	4	510 72 5 1 1
# 47	569	0.0	4	489 61 9 5 5
# 48	503	0.0	4	437 52 11 2 1
# 49	479	0.0	4	409 56 11 3
# 50	440	0.0	5	374 54 8 3 1
# 51	391	0.0	5	335 43 8 1 3 1
# 52	325	0.0	5	287 31 7
# 53	264	0.0	5	222 31 6 4 1
# 54	199	0.0	5	170 23 3 2 0 0 1
# 55	163	0.0	5	138 16 4 1 2 0 2
# 56	157	0.0	5	130 21 3 2 1
# 57	155	0.0	5	133 16 3 1 0 2
# 58	140	0.0	5	113 22 4 0 0 1
# 59	135	0.0	5	120 9 3 1 1 1
# 60	91	0.0	6	78 11 0 0 1 1
# 61	78	0.0	6	66 9 2 1
# 62	75	0.0	6	62 8 4 1
# 63	73	0.0	6	62 8 2 1
# 64	45	0.0	6	41 3 1
# 65	46	0.0	6	40 3 2 1
# 66	37	0.0	6	32 2 2 1
# 67	22	0.0	6	18 3 1
# 68	26	0.0	6	24 2
# 69	20	0.0	6	13 5 2
# 70	18	0.0	6	15 3
# 71	11	0.0	6	8 1 2
# 72	9	0.0	6	9
# 73	3	0.0	6	3
# 74	3	0.0	6	3
# 75	2	0.0	6	2
# 76	4	0.0	6	3 0 0 0 1
# 77	3	0.0	6	2 0 0 0 1
# 79	2	0.0	6	1 1
# 80	5	0.0	6	3 1 1
# 82	3	0.0	6	1 2
# 84	2	0.0	6	1 1
# 85	8	0.0	6	7 1
# 86	3	0.0	6	3
# 87	3	0.0	6	3
# 88	2	0.0	6	1 1
# 89	5	0.0	6	4 1
# 90	1	0.0	6	1
# 91	4	0.0	6	3 0 0 1
# 92	7	0.0	6	5 2
# 93	3	0.0	6	3
# 94	4	0.0	6	4
# 95	10	0.0	6	9 0 1
# 96	1	0.0	6	1
# 97	9	0.0	6	6 2 1
# 98	2	0.0	6	0 1 1
# 99	2	0.0	6	1 0 1
# 100	3420	0.0	6	1 2851 453 72 28 9 6
# 
# ----------------  L1_TGACCA_L001_R2_001.fastq.gz
# cutadapt version 1.4.2
# Command line parameters: -O 3 -e0.1 --too-short-output=L1_TGACCA_L001_R2_001.fastq.gz_shortReadsWithAdapter_2 -a AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT -m 50 --paired-output L1_TGACCA_L001_R1_001.fastq.gz.cutAdapters.fastq -o L1_TGACCA_L001_R2_001.fastq.gz.cutAdapters.fastq tmp.2.fastq tmp.1.fastq
# Maximum error rate: 10.00%
#    No. of adapters: 1
#    Processed reads:     21254588
#    Processed bases:   2125458800 bp (2125.5 Mbp)
#      Trimmed reads:      1295870 (6.1%)
#      Trimmed bases:     10729666 bp (10.7 Mbp) (0.50% of total)
#    Too short reads:            7 (0.0% of processed reads)
#     Too long reads:            0 (0.0% of processed reads)
#         Total time:    615.05 s
#      Time per read:      0.029 ms
# 
# === Adapter 1 ===
# 
# Adapter 'AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT', length 58, was trimmed 1295870 times.
# 
# No. of allowed errors:
# 0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-39 bp: 3; 40-49 bp: 4; 50-58 bp: 5
# 
# Overview of removed sequences
# length	count	expect	max.err	error counts
# 3	498867	332102.9	0	498867
# 4	148572	83025.7	0	148572
# 5	78904	20756.4	0	78904
# 6	57036	5189.1	0	57036
# 7	48657	1297.3	0	48657
# 8	43830	324.3	0	43830
# 9	40491	81.1	0	40005 486
# 10	37767	20.3	1	36663 1104
# 11	34078	5.1	1	33092 986
# 12	31279	1.3	1	30660 619
# 13	28986	0.3	1	28279 707
# 14	26813	0.1	1	26249 564
# 15	24882	0.0	1	24070 812
# 16	22163	0.0	1	21604 559
# 17	19785	0.0	1	19305 480
# 18	17066	0.0	1	16229 825 12
# 19	15063	0.0	1	14588 457 18
# 20	13368	0.0	2	12708 438 222
# 21	12148	0.0	2	11620 452 76
# 22	10626	0.0	2	10211 356 59
# 23	9473	0.0	2	9098 317 58
# 24	8760	0.0	2	8378 334 48
# 25	7967	0.0	2	7596 322 49
# 26	7462	0.0	2	7115 293 54
# 27	6422	0.0	2	6097 273 52
# 28	5561	0.0	2	5287 239 30 5
# 29	4847	0.0	2	4551 245 42 9
# 30	4281	0.0	3	4036 196 36 13
# 31	3897	0.0	3	3222 583 47 45
# 32	3323	0.0	3	3054 164 39 66
# 33	2900	0.0	3	2623 194 54 29
# 34	2573	0.0	3	2349 174 34 16
# 35	2286	0.0	3	2107 149 20 10
# 36	2021	0.0	3	1844 135 36 6
# 37	1916	0.0	3	1781 109 17 9
# 38	1817	0.0	3	1679 117 17 3 1
# 39	1533	0.0	3	1426 87 12 6 2
# 40	1429	0.0	4	1329 73 16 6 5
# 41	1217	0.0	4	1112 86 11 6 2
# 42	970	0.0	4	891 63 7 2 7
# 43	896	0.0	4	780 94 12 6 4
# 44	677	0.0	4	609 50 11 3 4
# 45	684	0.0	4	635 42 3 2 2
# 46	585	0.0	4	508 37 29 8 3
# 47	568	0.0	4	508 35 13 7 5
# 48	502	0.0	4	450 36 4 7 5
# 49	477	0.0	4	421 43 6 5 2
# 50	438	0.0	5	401 23 6 3 2 3
# 51	2	0.0	5	2
# 52	1	0.0	5	0 0 0 0 1
# 59	1	0.0	5	1
# 66	1	0.0	5	0 0 0 1
# 67	1	0.0	5	0 0 0 0 0 1
# 68	1	0.0	5	0 1
# 
# 
# --------------------------------------------------------
# --------------  Working on ACAGTG ----------------
# --------------------------------------------------------
# cutadapt version 1.4.2
# Command line parameters: -O 3 -e0.1 --too-short-output=L2_ACAGTG_L002_R1_001.fastq.gz_shortReadsWithAdapter_1 -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCACACAGTGATCTCGTATGCCGTCTTCTGCTTG -m 50 --paired-output tmp.2.fastq -o tmp.1.fastq L2_ACAGTG_L002_R1_001.fastq.gz L2_ACAGTG_L002_R2_001.fastq.gz
# Maximum error rate: 10.00%
#    No. of adapters: 1
#    Processed reads:     79014468
#    Processed bases:   7901446800 bp (7901.4 Mbp)
#      Trimmed reads:      3986125 (5.0%)
#      Trimmed bases:     30978284 bp (31.0 Mbp) (0.39% of total)
#    Too short reads:         9266 (0.0% of processed reads)
#     Too long reads:            0 (0.0% of processed reads)
#         Total time:   2451.37 s
#      Time per read:      0.031 ms
# 
# === Adapter 1 ===
# 
# Adapter 'AGATCGGAAGAGCACACGTCTGAACTCCAGTCACACAGTGATCTCGTATGCCGTCTTCTGCTTG', length 64, was trimmed 3986125 times.
# 
# No. of allowed errors:
# 0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-39 bp: 3; 40-49 bp: 4; 50-59 bp: 5; 60-64 bp: 6
# 
# Overview of removed sequences
# length	count	expect	max.err	error counts
# 3	1717614	1234601.1	0	1717614
# 4	483168	308650.3	0	483168
# 5	227944	77162.6	0	227944
# 6	157028	19290.6	0	157028
# 7	138362	4822.7	0	138362
# 8	124119	1205.7	0	124119
# 9	114021	301.4	0	112848 1173
# 10	103374	75.4	1	100239 3135
# 11	93772	18.8	1	90886 2886
# 12	85354	4.7	1	83478 1876
# 13	77493	1.2	1	75917 1576
# 14	71967	0.3	1	70351 1616
# 15	64858	0.1	1	63353 1505
# 16	58207	0.0	1	56713 1494
# 17	52394	0.0	1	51128 1266
# 18	46129	0.0	1	44908 1131 90
# 19	40643	0.0	1	39149 1392 102
# 20	36227	0.0	2	34956 1118 153
# 21	32567	0.0	2	31326 1091 150
# 22	28530	0.0	2	27421 1003 106
# 23	25762	0.0	2	24761 878 123
# 24	22993	0.0	2	22067 803 123
# 25	20953	0.0	2	20040 824 89
# 26	18849	0.0	2	18029 725 95
# 27	16763	0.0	2	16002 668 87 6
# 28	14299	0.0	2	13620 599 60 20
# 29	12767	0.0	2	12110 544 70 43
# 30	11248	0.0	3	10627 498 65 58
# 31	9522	0.0	3	8848 535 96 43
# 32	8639	0.0	3	8127 432 58 22
# 33	7655	0.0	3	7217 363 48 27
# 34	6791	0.0	3	6390 311 67 23
# 35	6121	0.0	3	5743 314 40 24
# 36	5601	0.0	3	5274 280 30 17
# 37	5038	0.0	3	4729 256 39 11 3
# 38	4441	0.0	3	4180 214 34 9 4
# 39	3941	0.0	3	3580 247 62 19 33
# 40	3469	0.0	4	3209 209 33 8 10
# 41	3006	0.0	4	2778 185 33 5 5
# 42	2491	0.0	4	2255 181 33 6 16
# 43	2300	0.0	4	2011 206 52 24 7
# 44	1938	0.0	4	1693 176 27 27 15
# 45	1755	0.0	4	1532 172 33 10 6 2
# 46	1559	0.0	4	1361 157 29 7 4 1
# 47	1553	0.0	4	1343 162 30 10 6 2
# 48	1348	0.0	4	1167 128 37 10 5 1
# 49	1254	0.0	4	1054 141 35 15 7 2
# 50	1032	0.0	5	903 100 17 8 2 2
# 51	935	0.0	5	799 98 24 10 3 1
# 52	735	0.0	5	637 76 14 6 2
# 53	644	0.0	5	561 65 11 4 1 2
# 54	564	0.0	5	479 67 9 3 1 1 4
# 55	510	0.0	5	438 47 17 5 2 0 1
# 56	369	0.0	5	316 37 6 5 3 2
# 57	406	0.0	5	335 49 10 4 4 1 3
# 58	288	0.0	5	253 30 4 0 0 1
# 59	327	0.0	5	273 37 10 1 3 3
# 60	251	0.0	6	207 30 7 2 3 0 2
# 61	224	0.0	6	184 29 4 3 3 1
# 62	164	0.0	6	141 15 3 4 0 0 1
# 63	157	0.0	6	134 15 4 0 2 0 2
# 64	127	0.0	6	110 13 1 1 1 1
# 65	138	0.0	6	117 16 1 2 1 0 1
# 66	75	0.0	6	59 13 3
# 67	95	0.0	6	80 13 0 0 1 0 1
# 68	75	0.0	6	58 8 4 3 2
# 69	70	0.0	6	59 6 1 1 1 1 1
# 70	50	0.0	6	38 7 1 1 2 1
# 71	52	0.0	6	44 5 2 1
# 72	38	0.0	6	35 2 1
# 73	40	0.0	6	33 3 2 2
# 74	27	0.0	6	24 2 1
# 75	34	0.0	6	31 2 1
# 76	36	0.0	6	28 6 1 1
# 77	29	0.0	6	19 7 1 1 0 1
# 78	34	0.0	6	25 6 0 1 0 0 2
# 79	33	0.0	6	29 2 2
# 80	45	0.0	6	31 11 2 1
# 81	39	0.0	6	34 2 1 0 0 1 1
# 82	49	0.0	6	41 6 1 0 1
# 83	54	0.0	6	35 10 2 5 1 0 1
# 84	29	0.0	6	21 3 4 1
# 85	51	0.0	6	44 3 2 1 1
# 86	74	0.0	6	61 8 1 2 1 1
# 87	57	0.0	6	45 8 0 0 3 1
# 88	76	0.0	6	70 4 1 0 0 1
# 89	50	0.0	6	37 7 3 2 0 0 1
# 90	54	0.0	6	41 9 1 2 1
# 91	106	0.0	6	83 13 4 2 2 0 2
# 92	80	0.0	6	68 8 2 1 1
# 93	86	0.0	6	73 9 2 1 1
# 94	41	0.0	6	30 4 4 0 2 0 1
# 95	133	0.0	6	113 13 6 1
# 96	54	0.0	6	46 4 1 0 2 1
# 97	267	0.0	6	228 28 6 3 0 2
# 98	80	0.0	6	59 8 4 0 4 3 2
# 99	25	0.0	6	18 4 0 0 0 2 1
# 100	1289	0.0	6	41 1016 156 30 10 19 17
# 
# ----------------  L2_ACAGTG_L002_R2_001.fastq.gz
# cutadapt version 1.4.2
# Command line parameters: -O 3 -e0.1 --too-short-output=L2_ACAGTG_L002_R2_001.fastq.gz_shortReadsWithAdapter_2 -a AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT -m 50 --paired-output L2_ACAGTG_L002_R1_001.fastq.gz.cutAdapters.fastq -o L2_ACAGTG_L002_R2_001.fastq.gz.cutAdapters.fastq tmp.2.fastq tmp.1.fastq
# Maximum error rate: 10.00%
#    No. of adapters: 1
#    Processed reads:     79005202
#    Processed bases:   7900520200 bp (7900.5 Mbp)
#      Trimmed reads:      4170338 (5.3%)
#      Trimmed bases:     30912967 bp (30.9 Mbp) (0.39% of total)
#    Too short reads:           41 (0.0% of processed reads)
#     Too long reads:            0 (0.0% of processed reads)
#         Total time:   2342.13 s
#      Time per read:      0.030 ms
# 
# === Adapter 1 ===
# 
# Adapter 'AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT', length 58, was trimmed 4170338 times.
# 
# No. of allowed errors:
# 0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-39 bp: 3; 40-49 bp: 4; 50-58 bp: 5
# 
# Overview of removed sequences
# length	count	expect	max.err	error counts
# 3	1853491	1234456.3	0	1853491
# 4	528617	308614.1	0	528617
# 5	243377	77153.5	0	243377
# 6	160172	19288.4	0	160172
# 7	138294	4822.1	0	138294
# 8	123502	1205.5	0	123502
# 9	113908	301.4	0	112347 1561
# 10	103544	75.3	1	99837 3707
# 11	93480	18.8	1	90422 3058
# 12	85115	4.7	1	83293 1822
# 13	76973	1.2	1	75042 1931
# 14	71557	0.3	1	70087 1470
# 15	64329	0.1	1	62058 2271
# 16	57758	0.0	1	56235 1523
# 17	52049	0.0	1	50719 1330
# 18	45641	0.0	1	43378 2218 45
# 19	40278	0.0	1	38987 1252 39
# 20	35970	0.0	2	34134 1166 670
# 21	32300	0.0	2	30869 1198 233
# 22	28339	0.0	2	27087 1075 177
# 23	25608	0.0	2	24486 934 188
# 24	22843	0.0	2	21871 839 133
# 25	20822	0.0	2	19729 958 135
# 26	18718	0.0	2	17768 831 119
# 27	16632	0.0	2	15801 702 129
# 28	14195	0.0	2	13419 640 134 2
# 29	12661	0.0	2	11825 713 110 13
# 30	11186	0.0	3	10527 530 89 40
# 31	9467	0.0	3	7838 1353 141 135
# 32	8585	0.0	3	7864 417 98 206
# 33	7607	0.0	3	6825 517 172 93
# 34	6735	0.0	3	6168 419 98 50
# 35	6067	0.0	3	5559 414 69 25
# 36	5573	0.0	3	5106 369 66 32
# 37	5009	0.0	3	4597 332 53 25 2
# 38	4413	0.0	3	4025 306 64 18
# 39	3915	0.0	3	3582 257 46 25 5
# 40	3448	0.0	4	3122 247 43 18 18
# 41	2997	0.0	4	2705 219 41 19 13
# 42	2476	0.0	4	2256 159 38 15 8
# 43	2291	0.0	4	1968 239 45 24 15
# 44	1921	0.0	4	1777 107 21 6 10
# 45	1735	0.0	4	1570 125 18 17 5
# 46	1547	0.0	4	1329 116 69 17 12 4
# 47	1542	0.0	4	1357 114 44 9 13 5
# 48	1342	0.0	4	1208 89 20 14 9 2
# 49	1239	0.0	4	1093 100 26 14 6
# 50	1029	0.0	5	902 92 18 4 3 10
# 51	2	0.0	5	0 0 0 1 0 1
# 52	3	0.0	5	2 0 0 0 1
# 53	2	0.0	5	0 1 0 0 1
# 54	3	0.0	5	2 0 0 0 0 1
# 55	4	0.0	5	1 1 1 0 0 1
# 56	2	0.0	5	1 0 0 0 0 1
# 57	2	0.0	5	2
# 58	2	0.0	5	0 1 0 0 1
# 60	1	0.0	5	1
# 61	2	0.0	5	2
# 65	2	0.0	5	0 0 0 1 1
# 66	2	0.0	5	0 0 1 0 0 1
# 70	1	0.0	5	0 0 0 0 0 1
# 71	1	0.0	5	1
# 72	1	0.0	5	0 0 0 0 0 1
# 75	1	0.0	5	1
# 76	1	0.0	5	1
# 79	1	0.0	5	0 0 0 0 1
# 84	1	0.0	5	0 1
# 86	2	0.0	5	1 0 0 0 0 1
# 95	1	0.0	5	0 0 1
# 97	2	0.0	5	0 1 0 0 0 1
# 98	1	0.0	5	0 0 0 0 1
# 100	1	0.0	5	0 0 0 1
# 
# --------------------------------------------------------
# --------------  Working on AGTCAA ----------------
# --------------------------------------------------------
# ----------------  L2_AGTCAA_L002_R1_001.fastq.gz
# cutadapt version 1.4.2
# Command line parameters: -O 3 -e0.1 --too-short-output=L2_AGTCAA_L002_R1_001.fastq.gz_shortReadsWithAdapter_1 -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCACAGTCAAATCTCGTATGCCGTCTTCTGCTTG -m 50 --paired-output tmp.2.fastq -o tmp.1.fastq L2_AGTCAA_L002_R1_001.fastq.gz L2_AGTCAA_L002_R2_001.fastq.gz
# Maximum error rate: 10.00%
#    No. of adapters: 1
#    Processed reads:     33547911
#    Processed bases:   3354791100 bp (3354.8 Mbp)
#      Trimmed reads:      1972839 (5.9%)
#      Trimmed bases:     17555768 bp (17.6 Mbp) (0.52% of total)
#    Too short reads:        11950 (0.0% of processed reads)
#     Too long reads:            0 (0.0% of processed reads)
#         Total time:   1030.51 s
#      Time per read:      0.031 ms
# 
# === Adapter 1 ===
# 
# Adapter 'AGATCGGAAGAGCACACGTCTGAACTCCAGTCACAGTCAAATCTCGTATGCCGTCTTCTGCTTG', length 64, was trimmed 1972839 times.
# 
# No. of allowed errors:
# 0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-39 bp: 3; 40-49 bp: 4; 50-59 bp: 5; 60-64 bp: 6
# 
# Overview of removed sequences
# length	count	expect	max.err	error counts
# 3	746669	524186.1	0	746669
# 4	213044	131046.5	0	213044
# 5	116583	32761.6	0	116583
# 6	85935	8190.4	0	85935
# 7	74958	2047.6	0	74958
# 8	68397	511.9	0	68397
# 9	63875	128.0	0	63467 408
# 10	59919	32.0	1	58132 1787
# 11	53354	8.0	1	52009 1345
# 12	48933	2.0	1	47816 1117
# 13	45341	0.5	1	44394 947
# 14	41180	0.1	1	40289 891
# 15	38191	0.0	1	37360 831
# 16	33651	0.0	1	32826 825
# 17	29932	0.0	1	29190 742
# 18	26124	0.0	1	25419 662 43
# 19	22757	0.0	1	21957 767 33
# 20	20650	0.0	2	19866 694 90
# 21	18364	0.0	2	17653 634 77
# 22	16464	0.0	2	15830 566 68
# 23	15072	0.0	2	14460 544 68
# 24	13790	0.0	2	13211 518 61
# 25	12444	0.0	2	11896 482 66
# 26	11315	0.0	2	10782 470 63
# 27	10380	0.0	2	9903 442 35
# 28	8723	0.0	2	8286 385 27 25
# 29	7643	0.0	2	7273 329 23 18
# 30	6802	0.0	3	6436 294 38 34
# 31	6061	0.0	3	5701 304 43 13
# 32	5271	0.0	3	4963 264 30 14
# 33	4780	0.0	3	4519 221 29 11
# 34	4171	0.0	3	3910 225 26 10
# 35	3700	0.0	3	3506 172 12 10
# 36	3495	0.0	3	3291 170 20 14
# 37	3266	0.0	3	3026 198 33 9
# 38	2904	0.0	3	2708 171 19 6
# 39	2572	0.0	3	2410 145 15 2
# 40	2353	0.0	4	2214 107 17 15
# 41	1968	0.0	4	6 16 1862 76 8
# 42	1628	0.0	4	0 1519 70 35 4
# 43	1417	0.0	4	0 4 1334 70 9
# 44	1207	0.0	4	0 0 1120 71 16
# 45	1078	0.0	4	0 3 963 92 20
# 46	1048	0.0	4	0 4 926 96 22
# 47	948	0.0	4	0 0 837 100 11
# 48	908	0.0	4	0 0 797 104 6 1
# 49	881	0.0	4	0 1 774 88 15 3
# 50	743	0.0	5	0 0 659 75 9
# 51	630	0.0	5	0 0 549 71 8 2
# 52	531	0.0	5	0 0 468 53 9 1
# 53	438	0.0	5	0 0 386 44 7 1
# 54	373	0.0	5	0 0 322 40 8 3
# 55	300	0.0	5	0 0 255 35 7 3
# 56	308	0.0	5	0 0 270 32 6
# 57	273	0.0	5	0 1 225 40 3 4
# 58	256	0.0	5	0 0 220 26 7 3
# 59	252	0.0	5	0 1 211 32 7 1
# 60	208	0.0	6	0 0 193 10 4 1
# 61	178	0.0	6	0 0 152 24 1 1
# 62	166	0.0	6	0 0 142 15 7 1 1
# 63	125	0.0	6	0 0 103 18 1 2 1
# 64	108	0.0	6	0 0 90 14 3 1
# 65	88	0.0	6	0 0 74 7 5 1 1
# 66	52	0.0	6	0 0 42 6 1 1 2
# 67	60	0.0	6	0 0 51 7 2
# 68	40	0.0	6	0 0 32 5 1 1 1
# 69	36	0.0	6	0 0 30 4 2
# 70	23	0.0	6	0 0 22 1
# 71	35	0.0	6	0 0 33 2
# 72	19	0.0	6	0 0 17 1 1
# 73	15	0.0	6	0 0 13 1 1
# 74	11	0.0	6	0 0 11
# 75	4	0.0	6	0 0 2 1 1
# 76	6	0.0	6	0 0 6
# 77	3	0.0	6	0 0 3
# 78	4	0.0	6	0 0 3 1
# 79	4	0.0	6	0 0 3 1
# 80	6	0.0	6	0 0 4 2
# 81	6	0.0	6	0 0 4 0 1 1
# 82	6	0.0	6	0 0 3 3
# 83	7	0.0	6	0 0 4 3
# 84	5	0.0	6	0 0 5
# 85	5	0.0	6	0 0 4 0 0 1
# 86	4	0.0	6	0 0 3 1
# 87	5	0.0	6	0 0 5
# 88	5	0.0	6	0 0 5
# 89	6	0.0	6	0 0 6
# 90	2	0.0	6	0 0 2
# 91	16	0.0	6	0 0 16
# 92	6	0.0	6	0 0 5 1
# 93	8	0.0	6	0 0 7 1
# 94	6	0.0	6	0 0 3 2 1
# 95	8	0.0	6	0 0 6 1 0 0 1
# 96	7	0.0	6	0 0 7
# 97	10	0.0	6	0 0 9 0 1
# 98	3	0.0	6	0 0 3
# 99	10	0.0	6	0 0 3 3 4
# 100	7273	0.0	6	1 3 1805 4682 606 130 46
# 
# ----------------  L2_AGTCAA_L002_R2_001.fastq.gz
# cutadapt version 1.4.2
# Command line parameters: -O 3 -e0.1 --too-short-output=L2_AGTCAA_L002_R2_001.fastq.gz_shortReadsWithAdapter_2 -a AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT -m 50 --paired-output L2_AGTCAA_L002_R1_001.fastq.gz.cutAdapters.fastq -o L2_AGTCAA_L002_R2_001.fastq.gz.cutAdapters.fastq tmp.2.fastq tmp.1.fastq
# Maximum error rate: 10.00%
#    No. of adapters: 1
#    Processed reads:     33535961
#    Processed bases:   3353596100 bp (3353.6 Mbp)
#      Trimmed reads:      2041468 (6.1%)
#      Trimmed bases:     16792909 bp (16.8 Mbp) (0.50% of total)
#    Too short reads:           33 (0.0% of processed reads)
#     Too long reads:            0 (0.0% of processed reads)
#         Total time:    991.27 s
#      Time per read:      0.030 ms
# 
# === Adapter 1 ===
# 
# Adapter 'AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT', length 58, was trimmed 2041468 times.
# 
# No. of allowed errors:
# 0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-39 bp: 3; 40-49 bp: 4; 50-58 bp: 5
# 
# Overview of removed sequences
# length	count	expect	max.err	error counts
# 3	799972	523999.4	0	799972
# 4	233522	130999.8	0	233522
# 5	124725	32750.0	0	124725
# 6	87633	8187.5	0	87633
# 7	74973	2046.9	0	74973
# 8	67995	511.7	0	67995
# 9	64179	127.9	0	63193 986
# 10	59861	32.0	1	58036 1825
# 11	53415	8.0	1	51606 1809
# 12	48822	2.0	1	47697 1125
# 13	45053	0.5	1	43867 1186
# 14	41023	0.1	1	40117 906
# 15	37853	0.0	1	36360 1493
# 16	33442	0.0	1	32513 929
# 17	29748	0.0	1	28892 856
# 18	25812	0.0	1	24394 1391 27
# 19	22588	0.0	1	21850 713 25
# 20	20465	0.0	2	19297 726 442
# 21	18226	0.0	2	17312 763 151
# 22	16406	0.0	2	15627 654 125
# 23	14973	0.0	2	14243 616 114
# 24	13712	0.0	2	13055 562 95
# 25	12368	0.0	2	11640 617 111
# 26	11248	0.0	2	10566 557 125
# 27	10305	0.0	2	9784 435 86
# 28	8661	0.0	2	8213 347 98 3
# 29	7588	0.0	2	7080 425 73 10
# 30	6776	0.0	3	6364 322 63 27
# 31	6025	0.0	3	4912 908 102 103
# 32	5228	0.0	3	4753 251 66 158
# 33	4748	0.0	3	4308 294 108 38
# 34	4130	0.0	3	3724 307 63 36
# 35	3675	0.0	3	3328 280 49 18
# 36	3474	0.0	3	3155 239 56 24
# 37	3237	0.0	3	2948 225 38 25 1
# 38	2878	0.0	3	2565 257 35 18 3
# 39	2552	0.0	3	2324 172 41 13 2
# 40	2342	0.0	4	2124 159 29 22 8
# 41	1960	0.0	4	1765 145 28 16 6
# 42	1618	0.0	4	1464 115 25 9 5
# 43	1412	0.0	4	1210 146 35 10 11
# 44	1205	0.0	4	1101 79 12 9 4
# 45	1078	0.0	4	960 84 10 16 8
# 46	1049	0.0	4	904 77 49 11 8
# 47	943	0.0	4	808 72 36 12 14 1
# 48	913	0.0	4	810 73 8 13 7 2
# 49	880	0.0	4	784 64 21 5 6
# 50	744	0.0	5	663 57 17 3 2 2
# 51	5	0.0	5	3 1 0 1
# 52	4	0.0	5	2 0 0 0 1 1
# 53	2	0.0	5	1 0 0 0 0 1
# 54	2	0.0	5	0 1 0 0 1
# 55	1	0.0	5	1
# 56	2	0.0	5	0 1 0 1
# 57	1	0.0	5	1
# 58	1	0.0	5	0 1
# 59	1	0.0	5	0 0 1
# 60	3	0.0	5	0 0 3
# 61	2	0.0	5	0 0 0 0 1 1
# 62	1	0.0	5	0 1
# 65	1	0.0	5	1
# 67	1	0.0	5	0 0 0 0 1
# 68	1	0.0	5	0 0 0 0 0 1
# 70	1	0.0	5	0 0 0 0 1
# 73	1	0.0	5	1
# 100	3	0.0	5	0 0 3
# 
# --------------------------------------------------------
# --------------  Working on CTTGTA ----------------
# --------------------------------------------------------
# ----------------  L2_CTTGTA_L002_R1_001.fastq.gz
# cutadapt version 1.4.2
# Command line parameters: -O 3 -e0.1 --too-short-output=L2_CTTGTA_L002_R1_001.fastq.gz_shortReadsWithAdapter_1 -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCACCTTGTAATCTCGTATGCCGTCTTCTGCTTG -m 50 --paired-output tmp.2.fastq -o tmp.1.fastq L2_CTTGTA_L002_R1_001.fastq.gz L2_CTTGTA_L002_R2_001.fastq.gz
# Maximum error rate: 10.00%
#    No. of adapters: 1
#    Processed reads:     19959731
#    Processed bases:   1995973100 bp (1996.0 Mbp)
#      Trimmed reads:      1105416 (5.5%)
#      Trimmed bases:      9617236 bp (9.6 Mbp) (0.48% of total)
#    Too short reads:         7360 (0.0% of processed reads)
#     Too long reads:            0 (0.0% of processed reads)
#         Total time:    609.49 s
#      Time per read:      0.031 ms
# 
# === Adapter 1 ===
# 
# Adapter 'AGATCGGAAGAGCACACGTCTGAACTCCAGTCACCTTGTAATCTCGTATGCCGTCTTCTGCTTG', length 64, was trimmed 1105416 times.
# 
# No. of allowed errors:
# 0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-39 bp: 3; 40-49 bp: 4; 50-59 bp: 5; 60-64 bp: 6
# 
# Overview of removed sequences
# length	count	expect	max.err	error counts
# 3	437905	311870.8	0	437905
# 4	123615	77967.7	0	123615
# 5	65260	19491.9	0	65260
# 6	46299	4873.0	0	46299
# 7	39960	1218.2	0	39960
# 8	36468	304.6	0	36468
# 9	33462	76.1	0	33191 271
# 10	31392	19.0	1	30427 965
# 11	28374	4.8	1	27576 798
# 12	25965	1.2	1	25328 637
# 13	24475	0.3	1	23948 527
# 14	22530	0.1	1	22000 530
# 15	20439	0.0	1	19944 495
# 16	18237	0.0	1	17715 522
# 17	15934	0.0	1	15530 404
# 18	13840	0.0	1	13440 366 34
# 19	12165	0.0	1	11719 416 30
# 20	11107	0.0	2	10721 336 50
# 21	9916	0.0	2	9518 353 45
# 22	8757	0.0	2	8386 332 39
# 23	8018	0.0	2	7667 308 43
# 24	7343	0.0	2	7047 257 39
# 25	6771	0.0	2	6457 265 49
# 26	6098	0.0	2	5802 265 31
# 27	5325	0.0	2	5108 188 26 3
# 28	4746	0.0	2	4501 216 26 3
# 29	4060	0.0	2	3858 171 18 13
# 30	3631	0.0	3	3433 163 16 19
# 31	2994	0.0	3	2783 173 27 11
# 32	2801	0.0	3	2635 139 22 5
# 33	2481	0.0	3	2369 100 8 4
# 34	2095	0.0	3	1969 103 17 6
# 35	2038	0.0	3	1910 110 13 5
# 36	1862	0.0	3	1778 67 10 7
# 37	1658	0.0	3	1566 77 13 2
# 38	1481	0.0	3	1394 73 11 3
# 39	1375	0.0	3	1267 83 14 4 7
# 40	1222	0.0	4	1138 71 8 5
# 41	1005	0.0	4	940 53 10 2
# 42	839	0.0	4	771 54 7 4 3
# 43	737	0.0	4	647 74 9 7
# 44	582	0.0	4	518 44 6 8 6
# 45	545	0.0	4	483 51 6 3 1 1
# 46	503	0.0	4	457 39 4 0 3
# 47	477	0.0	4	418 40 7 5 6 1
# 48	478	0.0	4	426 43 7 1 1
# 49	420	0.0	4	379 30 6 2 1 2
# 50	371	0.0	5	339 26 4 1 1
# 51	337	0.0	5	303 27 5 1 1
# 52	261	0.0	5	229 24 2 3 2 1
# 53	211	0.0	5	182 22 5 2
# 54	175	0.0	5	160 10 3 1 1
# 55	147	0.0	5	132 11 1 2 0 0 1
# 56	139	0.0	5	125 9 4 0 1
# 57	140	0.0	5	122 12 4 1 1
# 58	121	0.0	5	107 11 0 1 1 0 1
# 59	112	0.0	5	97 14 1
# 60	117	0.0	6	100 9 4 1 2 1
# 61	93	0.0	6	82 7 4
# 62	76	0.0	6	64 12
# 63	67	0.0	6	52 15
# 64	54	0.0	6	46 7 0 0 1
# 65	49	0.0	6	44 2 3
# 66	37	0.0	6	32 3 1 1
# 67	25	0.0	6	22 3
# 68	23	0.0	6	19 3 0 1
# 69	30	0.0	6	23 4 0 1 1 0 1
# 70	17	0.0	6	15 0 1 1
# 71	12	0.0	6	9 1 1 0 0 0 1
# 72	10	0.0	6	8 2
# 73	8	0.0	6	7 0 1
# 74	10	0.0	6	10
# 75	5	0.0	6	4 0 0 1
# 76	4	0.0	6	4
# 77	1	0.0	6	1
# 78	3	0.0	6	1 1 0 0 0 1
# 79	3	0.0	6	2 0 0 0 1
# 80	3	0.0	6	3
# 81	1	0.0	6	1
# 82	5	0.0	6	5
# 83	3	0.0	6	2 1
# 84	2	0.0	6	1 0 1
# 85	2	0.0	6	2
# 86	1	0.0	6	1
# 88	3	0.0	6	3
# 89	2	0.0	6	2
# 90	5	0.0	6	4 1
# 91	5	0.0	6	5
# 92	6	0.0	6	5 1
# 93	4	0.0	6	3 1
# 94	3	0.0	6	3
# 95	7	0.0	6	6 1
# 96	5	0.0	6	5
# 97	12	0.0	6	9 3
# 98	3	0.0	6	3
# 99	2	0.0	6	0 1 1
# 100	4999	0.0	6	240 4127 477 105 36 9 5
# 
# ----------------  L2_CTTGTA_L002_R2_001.fastq.gz
# cutadapt version 1.4.2
# Command line parameters: -O 3 -e0.1 --too-short-output=L2_CTTGTA_L002_R2_001.fastq.gz_shortReadsWithAdapter_2 -a AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT -m 50 --paired-output L2_CTTGTA_L002_R1_001.fastq.gz.cutAdapters.fastq -o L2_CTTGTA_L002_R2_001.fastq.gz.cutAdapters.fastq tmp.2.fastq tmp.1.fastq
# Maximum error rate: 10.00%
#    No. of adapters: 1
#    Processed reads:     19952371
#    Processed bases:   1995237100 bp (1995.2 Mbp)
#      Trimmed reads:      1150320 (5.8%)
#      Trimmed bases:      9129502 bp (9.1 Mbp) (0.46% of total)
#    Too short reads:           13 (0.0% of processed reads)
#     Too long reads:            0 (0.0% of processed reads)
#         Total time:    586.92 s
#      Time per read:      0.029 ms
# 
# === Adapter 1 ===
# 
# Adapter 'AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT', length 58, was trimmed 1150320 times.
# 
# No. of allowed errors:
# 0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-39 bp: 3; 40-49 bp: 4; 50-58 bp: 5
# 
# Overview of removed sequences
# length	count	expect	max.err	error counts
# 3	475217	311755.8	0	475217
# 4	135347	77938.9	0	135347
# 5	69619	19484.7	0	69619
# 6	47202	4871.2	0	47202
# 7	40059	1217.8	0	40059
# 8	36173	304.4	0	36173
# 9	33405	76.1	0	33021 384
# 10	31359	19.0	1	30301 1058
# 11	28404	4.8	1	27411 993
# 12	25870	1.2	1	25267 603
# 13	24273	0.3	1	23607 666
# 14	22417	0.1	1	21931 486
# 15	20263	0.0	1	19504 759
# 16	18070	0.0	1	17550 520
# 17	15829	0.0	1	15447 382
# 18	13681	0.0	1	13010 660 11
# 19	12079	0.0	1	11684 387 8
# 20	11024	0.0	2	10405 383 236
# 21	9850	0.0	2	9368 390 92
# 22	8693	0.0	2	8306 320 67
# 23	7970	0.0	2	7588 317 65
# 24	7314	0.0	2	7002 252 60
# 25	6722	0.0	2	6390 289 43
# 26	6040	0.0	2	5716 268 56
# 27	5287	0.0	2	5023 221 43
# 28	4697	0.0	2	4467 181 46 3
# 29	4019	0.0	2	3762 207 44 6
# 30	3616	0.0	3	3398 167 32 19
# 31	2976	0.0	3	2457 402 61 56
# 32	2784	0.0	3	2554 121 30 79
# 33	2469	0.0	3	2227 168 44 30
# 34	2079	0.0	3	1903 132 30 14
# 35	2021	0.0	3	1838 140 27 16
# 36	1843	0.0	3	1684 119 28 12
# 37	1647	0.0	3	1521 92 24 10
# 38	1473	0.0	3	1352 99 17 5
# 39	1368	0.0	3	1241 95 18 13 1
# 40	1214	0.0	4	1089 99 13 8 5
# 41	1004	0.0	4	909 68 13 8 6
# 42	835	0.0	4	762 55 11 4 3
# 43	727	0.0	4	624 82 6 8 7
# 44	581	0.0	4	526 40 10 3 2
# 45	543	0.0	4	494 37 8 3 1
# 46	504	0.0	4	433 31 27 6 4 3
# 47	476	0.0	4	411 44 9 7 4 1
# 48	476	0.0	4	436 24 8 4 3 1
# 49	419	0.0	4	375 29 6 5 4
# 50	369	0.0	5	336 22 6 2 1 2
# 51	2	0.0	5	1 0 0 1
# 53	2	0.0	5	0 0 1 0 0 1
# 54	1	0.0	5	0 1
# 56	1	0.0	5	0 0 0 0 0 1
# 57	1	0.0	5	0 0 0 0 0 1
# 59	1	0.0	5	0 0 0 1
# 61	1	0.0	5	0 0 0 0 0 1
# 62	1	0.0	5	0 1
# 77	1	0.0	5	0 0 1
# 100	2	0.0	5	1 0 0 0 0 1
# 
# --------------------------------------------------------
# --------------  Working on GTGAAA ----------------
# --------------------------------------------------------
# ----------------  L2_GTGAAA_L002_R1_001.fastq.gz
# cutadapt version 1.4.2
# Command line parameters: -O 3 -e0.1 --too-short-output=L2_GTGAAA_L002_R1_001.fastq.gz_shortReadsWithAdapter_1 -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCACGTGAAAATCTCGTATGCCGTCTTCTGCTTG -m 50 --paired-output tmp.2.fastq -o tmp.1.fastq L2_GTGAAA_L002_R1_001.fastq.gz L2_GTGAAA_L002_R2_001.fastq.gz
# Maximum error rate: 10.00%
#    No. of adapters: 1
#    Processed reads:     30777541
#    Processed bases:   3077754100 bp (3077.8 Mbp)
#      Trimmed reads:      2155163 (7.0%)
#      Trimmed bases:     20883042 bp (20.9 Mbp) (0.68% of total)
#    Too short reads:        10672 (0.0% of processed reads)
#     Too long reads:            0 (0.0% of processed reads)
#         Total time:    950.56 s
#      Time per read:      0.031 ms
# 
# === Adapter 1 ===
# 
# Adapter 'AGATCGGAAGAGCACACGTCTGAACTCCAGTCACGTGAAAATCTCGTATGCCGTCTTCTGCTTG', length 64, was trimmed 2155163 times.
# 
# No. of allowed errors:
# 0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-39 bp: 3; 40-49 bp: 4; 50-59 bp: 5; 60-64 bp: 6
# 
# Overview of removed sequences
# length	count	expect	max.err	error counts
# 3	697661	480899.1	0	697661
# 4	218802	120224.8	0	218802
# 5	128609	30056.2	0	128609
# 6	98842	7514.0	0	98842
# 7	88360	1878.5	0	88360
# 8	80348	469.6	0	80348
# 9	75389	117.4	0	74951 438
# 10	70871	29.4	1	68953 1918
# 11	64272	7.3	1	62658 1614
# 12	59220	1.8	1	57921 1299
# 13	56526	0.5	1	55363 1163
# 14	51792	0.1	1	50578 1214
# 15	48518	0.0	1	47394 1124
# 16	43579	0.0	1	42525 1054
# 17	39253	0.0	1	38349 904
# 18	34396	0.0	1	33490 844 62
# 19	30724	0.0	1	29609 1030 85
# 20	27685	0.0	2	26730 824 131
# 21	24920	0.0	2	23974 819 127
# 22	22243	0.0	2	21377 754 112
# 23	20289	0.0	2	19543 679 67
# 24	18438	0.0	2	17720 642 76
# 25	16906	0.0	2	16220 621 65
# 26	15209	0.0	2	14559 560 90
# 27	13829	0.0	2	13250 508 69 2
# 28	12101	0.0	2	11577 461 51 12
# 29	10238	0.0	2	9769 379 57 33
# 30	9311	0.0	3	8839 387 41 44
# 31	8007	0.0	3	7553 363 62 29
# 32	7238	0.0	3	6849 337 46 6
# 33	6444	0.0	3	6094 300 38 12
# 34	5618	0.0	3	5327 251 27 13
# 35	4991	0.0	3	4720 229 26 16
# 36	4568	0.0	3	4320 197 38 13
# 37	4150	0.0	3	3922 193 23 12
# 38	3809	0.0	3	3607 180 17 4 1
# 39	3405	0.0	3	3199 175 17 10 4
# 40	3038	0.0	4	2852 140 16 25 5
# 41	2461	0.0	4	5 28 2326 73 29
# 42	2073	0.0	4	0 6 4 30 2033
# 43	1863	0.0	4	0 0 1753 68 42
# 44	1587	0.0	4	0 2 1445 109 31
# 45	1423	0.0	4	0 0 1229 168 21 5
# 46	1283	0.0	4	0 2 1087 159 32 3
# 47	1201	0.0	4	0 1 1010 172 18
# 48	1068	0.0	4	0 2 916 129 21
# 49	1022	0.0	4	0 0 854 135 30 3
# 50	911	0.0	5	0 1 769 116 25
# 51	757	0.0	5	0 1 647 90 17 2
# 52	668	0.0	5	0 0 572 80 11 5
# 53	559	0.0	5	0 0 480 70 5 4
# 54	433	0.0	5	0 0 367 52 12 2
# 55	376	0.0	5	0 1 317 47 8 3
# 56	306	0.0	5	0 0 260 39 4 3
# 57	336	0.0	5	0 1 288 37 8 2
# 58	312	0.0	5	0 0 267 38 7
# 59	288	0.0	5	0 0 243 37 7 1
# 60	200	0.0	6	0 1 167 28 3 1
# 61	192	0.0	6	0 0 154 27 6 5
# 62	162	0.0	6	0 0 132 25 2 0 3
# 63	123	0.0	6	0 0 103 16 1 3
# 64	104	0.0	6	0 0 81 17 3 3
# 65	92	0.0	6	0 0 75 13 2 2
# 66	88	0.0	6	0 0 76 8 2 1 1
# 67	55	0.0	6	0 1 43 9 1 1
# 68	62	0.0	6	0 0 53 7 0 1 1
# 69	49	0.0	6	0 0 34 8 5 2
# 70	35	0.0	6	0 0 28 2 3 1 1
# 71	25	0.0	6	0 0 21 4
# 72	12	0.0	6	0 0 10 2
# 73	18	0.0	6	0 0 16 1 1
# 74	11	0.0	6	0 0 9 1 1
# 75	7	0.0	6	0 0 4 1 2
# 76	11	0.0	6	0 0 9 1 0 0 1
# 77	8	0.0	6	0 0 7 1
# 78	4	0.0	6	0 0 4
# 79	5	0.0	6	0 0 3 1 0 1
# 80	4	0.0	6	0 0 2 2
# 81	3	0.0	6	0 0 3
# 82	3	0.0	6	0 0 3
# 83	7	0.0	6	0 0 4 1 1 1
# 84	5	0.0	6	0 0 4 1
# 85	2	0.0	6	0 0 2
# 86	8	0.0	6	0 0 6 2
# 87	6	0.0	6	0 0 6
# 88	3	0.0	6	0 0 3
# 89	7	0.0	6	0 0 5 2
# 90	4	0.0	6	0 0 4
# 91	10	0.0	6	0 0 10
# 92	6	0.0	6	0 0 5 1
# 93	8	0.0	6	0 0 7 1
# 94	1	0.0	6	0 0 1
# 95	9	0.0	6	0 0 8 0 1
# 96	3	0.0	6	0 0 3
# 97	12	0.0	6	0 0 11 1
# 98	3	0.0	6	0 0 2 1
# 99	1	0.0	6	0 0 1
# 100	5269	0.0	6	0 0 6 4380 700 133 50
# 
# ----------------  L2_GTGAAA_L002_R2_001.fastq.gz
# cutadapt version 1.4.2
# Command line parameters: -O 3 -e0.1 --too-short-output=L2_GTGAAA_L002_R2_001.fastq.gz_shortReadsWithAdapter_2 -a AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT -m 50 --paired-output L2_GTGAAA_L002_R1_001.fastq.gz.cutAdapters.fastq -o L2_GTGAAA_L002_R2_001.fastq.gz.cutAdapters.fastq tmp.2.fastq tmp.1.fastq
# Maximum error rate: 10.00%
#    No. of adapters: 1
#    Processed reads:     30766869
#    Processed bases:   3076686900 bp (3076.7 Mbp)
#      Trimmed reads:      2223495 (7.2%)
#      Trimmed bases:     20256509 bp (20.3 Mbp) (0.66% of total)
#    Too short reads:           46 (0.0% of processed reads)
#     Too long reads:            0 (0.0% of processed reads)
#         Total time:    913.65 s
#      Time per read:      0.030 ms
# 
# === Adapter 1 ===
# 
# Adapter 'AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT', length 58, was trimmed 2223495 times.
# 
# No. of allowed errors:
# 0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-39 bp: 3; 40-49 bp: 4; 50-58 bp: 5
# 
# Overview of removed sequences
# length	count	expect	max.err	error counts
# 3	755209	480732.3	0	755209
# 4	236485	120183.1	0	236485
# 5	135321	30045.8	0	135321
# 6	100304	7511.4	0	100304
# 7	88338	1877.9	0	88338
# 8	79776	469.5	0	79776
# 9	75354	117.4	0	74750 604
# 10	70653	29.3	1	68770 1883
# 11	64296	7.3	1	62422 1874
# 12	59036	1.8	1	57803 1233
# 13	56190	0.5	1	54800 1390
# 14	51553	0.1	1	50532 1021
# 15	48202	0.0	1	46502 1700
# 16	43263	0.0	1	42142 1121
# 17	39031	0.0	1	38050 981
# 18	34094	0.0	1	32350 1724 20
# 19	30524	0.0	1	29601 893 30
# 20	27456	0.0	2	26089 854 513
# 21	24738	0.0	2	23658 926 154
# 22	22121	0.0	2	21186 802 133
# 23	20192	0.0	2	19366 691 135
# 24	18359	0.0	2	17561 670 128
# 25	16786	0.0	2	15978 679 129
# 26	15127	0.0	2	14370 641 116
# 27	13738	0.0	2	13048 566 124
# 28	12030	0.0	2	11432 475 118 5
# 29	10174	0.0	2	9525 555 76 18
# 30	9262	0.0	3	8722 414 84 42
# 31	7977	0.0	3	6580 1137 126 134
# 32	7197	0.0	3	6646 298 76 177
# 33	6400	0.0	3	5810 391 141 58
# 34	5582	0.0	3	5143 329 73 37
# 35	4945	0.0	3	4541 319 63 22
# 36	4543	0.0	3	4145 316 57 25
# 37	4130	0.0	3	3788 261 57 24
# 38	3788	0.0	3	3490 224 48 26
# 39	3377	0.0	3	3094 227 38 15 3
# 40	3022	0.0	4	2780 186 33 17 6
# 41	2453	0.0	4	2264 146 30 11 2
# 42	2059	0.0	4	1885 132 31 9 2
# 43	1863	0.0	4	1587 206 45 14 11
# 44	1581	0.0	4	1449 94 20 11 7
# 45	1424	0.0	4	1296 95 16 10 7
# 46	1276	0.0	4	1129 73 56 10 8
# 47	1202	0.0	4	1079 78 28 10 7
# 48	1069	0.0	4	963 71 16 13 6
# 49	1027	0.0	4	917 75 18 7 8 2
# 50	922	0.0	5	822 69 15 7 3 6
# 51	8	0.0	5	4 2 0 1 1
# 52	9	0.0	5	2 0 3 0 3 1
# 53	2	0.0	5	1 0 0 0 0 1
# 55	4	0.0	5	0 2 1 0 1
# 56	2	0.0	5	1 1
# 57	2	0.0	5	0 0 1 0 0 1
# 58	3	0.0	5	1 2
# 59	4	0.0	5	0 1 1 0 1 1
# 63	2	0.0	5	0 2
# 66	2	0.0	5	0 0 0 0 0 2
# 70	1	0.0	5	1
# 72	3	0.0	5	1 0 1 0 0 1
# 74	1	0.0	5	0 0 1
# 76	1	0.0	5	0 0 0 0 0 1
# 77	1	0.0	5	0 0 1
# 97	1	0.0	5	0 1


----------------  L2_ACAGTG_L002_R1_001.fastq.gz
